dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -461,11 +462,14 @@ private void scanForEndpoints() throws CDIException {
                     for (ManagedBeanDescriptor<?> managedBeanDescriptor : managedBeanDescriptors) {
                         boolean added = false;
                         for (WebSphereBeanDeploymentArchive child : getWebSphereBeanDeploymentArchives()) {
-                            if ((child.getAllClazzes().containsKey(managedBeanDescriptor.getBeanClass().getName()))
-                                && (child.getAllClazzes().containsValue(managedBeanDescriptor.getBeanClass()))) {
-                                child.addManagedBeanDescriptor(managedBeanDescriptor);
-                                added = true;
-                                break;
+                            if (child.getAllClazzes().contains(managedBeanDescriptor.getBeanClass().getName())) {
```

Comment:
Is it safe to only check by name? I think you need to check that the classloader matches too?